lib/medium-client.js

```diff
@@ -0,0 +1,423 @@
+ *
+ * Note that following restrictions around this function:
+ * - Current user must be an editor or a writer for the specified publication
+ * - If user is an editor, they may request any publishStatus. The post will
+ *   be directly approved for the specified publication
+ * - If user is a writer, they may only request 'draft' as publishStatus. The
+ *   post will appear in the publication as 'pending approval', visible only
+ *   to the editor(s) of the publication
+ *
+ * Requires the current access token to have the publishPost scope.
+ *
+ * @param {{
+ *  userId: string,
+ *  publicationId: string,
+ *  title: string,
+ *  contentFormat: PostContentFormat,
+ *  content: string,
+ *  tags: Array.<string>,
+ *  canonicalUrl: string,
+ *  publishStatus: PostPublishStatus,
+ *  license: PostLicense
+ * }} options
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype.createPostInPublication = function (options, callback) {
+  this._enforce(options, ['publicationId', 'title', 'contentFormat', 'content'])
+  this._makeRequest({
+    method: 'POST',
+    path: '/v1/publications/' + options.publicationId + '/posts',
+    data: {
+      title: options.title,
+      content: options.content,
+      contentFormat: options.contentFormat,
+      tags: options.tags,
+      canonicalUrl: options.canonicalUrl,
+      publishStatus: options.publishStatus,
+      license: options.license
+    }
+  }, callback)
+}
+
+
+/**
+ * Acquires an access token for the Medium API.
+ *
+ * Sets the access token on the client on success.
+ *
+ * @param {Object} params
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype._acquireAccessToken = function (params, callback) {
+  this._makeRequest({
+    method: 'POST',
+    path: '/v1/tokens',
+    contentType: 'application/x-www-form-urlencoded',
+    data: qs.stringify(params)
+  }, function (err, data) {
+    if (!err) {
+      this._accessToken = data.accessToken
+    }
+    callback(err, data)
+  }.bind(this))
+}
+
+
+/**
+ * Enforces that given options object (first param) defines
+ * all keys requested (second param). Raises an error if any
+ * is missing.
+ *
+ * @param {Object} options
+ * @param {keys} requiredKeys
+ */
+MediumClient.prototype._enforce = function (options, requiredKeys) {
+  if (!options) {
+    throw new MediumError('Parameters for this call are undefined', MISSING_REQUIRED_PARAM)
+  }
+  requiredKeys.forEach(function (requiredKey) {
+    if (!options[requiredKey]) throw new MediumError('Missing required parameter "' + requiredKey + '"', MISSING_REQUIRED_PARAM)
+  })
+}
+
+
+
+/**
+ * Makes a request to the Medium API.
+ *
+ * @param {Object} options
+ * @param {NodeCallback} callback
+ */
+MediumClient.prototype._makeRequest = function (options, callback) {
+  // The bypass is for testing purposes only - api.medium.com will
+  // not accept HTTP connections
+  var useSsl = !process.env.MEDIUM_API_USE_HTTP
```

Comment:
I don't see a problem here. Adding this flag does not introduce any kind of security vulnerability anywhere - anyone interested in making these requests over HTTP could change the code very easily, and this shouldn't accidentally turn HTTP on for anyone. What this gives us though is possibility of running integration tests against standard.dev:8080, which has obvious value to it.